tests/apple-pay/applepay.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { test, CustomerStepPreset, PaymentStepAsGuestPreset } from '../';
+import { internalOrder, order, orderPayment, validateMerchantResponse, consignments } from './ApplePayTestMockResponse';
+
+test.describe('ApplePay', () => {
+    test.skip('ApplePay in payment step', async ({ assertions, checkout, page }) => {
+        // Testing environment setup
+        await page.addInitScript({ path: './tests/apple-pay/ApplePaySessionMockObject.js' });
+
+        const storeUrl = 'https://my-dev-store-759854325.store.bcdev';
+        const responseProps = { status: 200, contentType: 'application/json' };
+        await checkout.use(new PaymentStepAsGuestPreset(storeUrl));
+        await checkout.create('ApplePay in Payment Step', storeUrl);
+        await checkout.route(/order-confirmation.*/, './tests/support/orderConfirmation.ejs', { orderId: '124' });
+        await page.route('**/api/public/v1/payments/applepay/validate_merchant', route => route.fulfill({ ...responseProps, body: validateMerchantResponse }));
+        await page.route(/.*\/api\/storefront\/orders\/124.*/, route => route.fulfill({...responseProps, body: order }));
+        await page.route('**/api/public/v1/orders/payments', route => route.fulfill({ ...responseProps, body: orderPayment }));
+        await page.route('**/internalapi/v1/checkout/order', route => route.fulfill({
+            ...responseProps,
+            headers: { Token: 'White shirt now red, my bloody nose' },
+            body: internalOrder,
+        }));
+
+        // Playwright actions
+        await checkout.goto();
+        await page.locator('label div').first().click();
+        await checkout.placeOrder();
+
+        // Assertions
+        await assertions.shouldSeeOrderConfirmation();        
+    })
+
+    test('ApplePay in customer step', async ({ assertions, checkout, page }) => {
+        // Testing environment setup
+        await page.addInitScript({ path: './tests/apple-pay/ApplePaySessionMockObject.js' });
+
+        const storeUrl = 'https://my-dev-store-759854325.store.bcdev';
+        const responseProps = { status: 200, contentType: 'application/json' };
+        await checkout.use(new CustomerStepPreset(storeUrl));
+        await checkout.create('ApplePay in Customer Step', storeUrl);        
+
+        await checkout.route(/order-confirmation.*/, './tests/support/orderConfirmation.ejs', { orderId: '124' });
+        await page.route('**/api/public/v1/payments/applepay/validate_merchant', route => route.fulfill({ ...responseProps, body: validateMerchantResponse }));        
+        await page.route(/.*\/api\/storefront\/orders\/124.*/, route => route.fulfill({...responseProps, body: order }));
+        await page.route('**/api/public/v1/orders/payments', route => route.fulfill({ ...responseProps, body: orderPayment }));
+        await page.route(/.*\/api\/storefront\/checkouts\/124.*\/consignments/, route => route.fulfill({...responseProps, body: consignments }));
```

Comment:
It appears this test does not make calls to update shipping and contact. I see there is only a validating merchant event on line 82 of `ApplePaySessionMockObject.js`. Perhaps we can make another `ApplePaySessionMockObject` for this test.

Reply:
I've added the consignment and the billing address mock routes to match the shipping and the contact function calls. Is this adequate?

Reply:
> 1. `this.onvalidatemerchant(event);` on [ApplePaySessionMockObject.js](https://github.com/bigcommerce/checkout-js/blob/e0f10d9a34ff3add5562be228f169424713f1c96/tests/apple-pay/ApplePaySessionMockObject.js#L63) will run the event defined in `checkout-sdk`.
> 2. API calls will be made by the actual code we want to test.
> 3. `playwright` will intercept the call and return mocked data.

It looks like you have finished step 3. Step 1 and 2 are still missing. Hence the test does not cover the code we want to test.